# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/websocket/WebSocket.cpp
# Commit: ffc69aa8f2c3
# Full Hash: ffc69aa8f2c3bcdc3cc4f2516a618c0430e67ddf
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2024-04-19 15:22:44
# Regressor Bug: 1769913
# File Overlap Count: 3
# Description:
#   Bug 1769913 - P2 WorkerParentThreadRunnable for the runnables dispatched to worker's parent thread. r=dom-worker-reviewers,asuth
#   
#   In this patch, WorkerParentThreadRunnable is extracted from WorkerThreadRunnable for runnable on the parent thread.
#   
#   WorkerParentControlRunnable and WorkerParentDebuggeeRunnable are also created for control runnable and debuggee runnable on the parent thread.
# ==============================================================================

diff -r 379cd5044768 -r ffc69aa8f2c3 dom/websocket/WebSocket.cpp
--- a/dom/websocket/WebSocket.cpp	Fri Apr 19 09:41:57 2024 +0000
+++ b/dom/websocket/WebSocket.cpp	Fri Apr 19 09:41:57 2024 +0000
@@ -2792,8 +2792,7 @@
   WorkerRunnableDispatcher(WebSocketImpl* aImpl,
                            ThreadSafeWorkerRef* aWorkerRef,
                            already_AddRefed<nsIRunnable> aEvent)
-      : WorkerThreadRunnable(aWorkerRef->Private(), "WorkerRunnableDispatcher",
-                             WorkerRunnable::WorkerThread),
+      : WorkerThreadRunnable(aWorkerRef->Private(), "WorkerRunnableDispatcher"),
         mWebSocketImpl(aImpl),
         mEvent(std::move(aEvent)) {}
 